File: backend/tiggle/src/main/java/com/side/tiggle/domain/grade/model/Grade.java
package com.side.tiggle.domain.grade.model;

import com.side.tiggle.domain.reaction.model.ReactionType;
import com.side.tiggle.global.common.model.BaseEntity;
import lombok.*;

import javax.persistence.*;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "grades")
public class Grade extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "image_url", nullable = false)
    private String imageUrl;

    @Column(name = "description", nullable = false)
    private String description;

    @Builder
    public Grade(String name, String imageUrl, String description) {
        this.name = name;
        this.imageUrl = imageUrl;
        this.description = description;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/grade/model/Grade.kt
package com.side.tiggle.domain.grade.model

import com.side.tiggle.global.common.model.BaseEntity
import javax.persistence.*

@Entity
@Table(name = "grades")
class Grade(
    @Column(name = "name", nullable = false)
    var name: String,

    @Column(name = "image_url", nullable = false)
    var imageUrl: String,

    @Column(name = "description", nullable = false)
    var description: String
): BaseEntity() {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0
}